src/lib/molecules/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,604 @@
+    return value.length <= 2 && Array.isArray(format)
+        ? Number(value) < 12
+        : (isLongHour(format) && Number(value) < 24) || (isShortHour(format) && Number(value) < 12);
+};
+
+const checkRange = (value: string) => value.length <= 2 && Number(value) < 60;
+
+const isLongHour = (str: string | string[]) => str === 'HH' || str === 'H';
+const isShortHour = (str: string | string[]) => str === 'hh' || str === 'h';
+
+function convertToFormat(value: string, format: string | string[] | undefined, notEmpty: boolean = true) {
+    if (!value && notEmpty) {
+        return format?.length === 1 ? '0' : '00';
+    }
+    if (format?.length === 2 && value?.length === 1) {
+        return `0${value}`;
+    }
+    if (format?.length === 1 && value?.length === 2) {
+        return Number(value).toString();
+    }
+
+    return value;
+}
+
+interface ITimePickerProps {
+    /**
+     * Define is seconds field will be shown or no
+     */
+    showSeconds?: boolean;
+    /**
+     * Select view with multiple inputs or with single inputs <br>
+     * Possible values: `'multipleInputs' | 'singleInput'`
+     */
+    appearance?: 'multipleInputs' | 'singleInput';
+    /**
+     * Format for hour field <br>
+     * Possible values: `'HH' | 'H' | 'hh' | 'h' | ['HH', 'H', 'hh', 'h']`
+     */
+    hourFormat?: 'HH' | 'H' | 'hh' | 'h' | string[];
+    /**
+     * Format for hour field <br>
+     * Possible values: `'mm' | 'm' | ['mm' | 'm']`
+     */
+    minuteFormat?: 'mm' | 'm' | string[];
+    /**
+     * Format for hour field <br>
+     * Possible values: `'ss' | 's' | ['ss' | 's']`
+     */
+    secondFormat?: 'ss' | 's' | string[];
+    /**
+     * Time field separator
+     */
+    separator?: string;
+    /**
+     * Fires an event when field is changed
+     * `((event: SyntheticEvent) => void)`.
+     */
+    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+    /**
+     * Fires an event when field is blurred
+     * `((date: Date, event: SyntheticEvent) => void)`.
+     */
+    onBlur?: (e: React.FocusEvent<HTMLInputElement>) => void;
+    /**
+     * Additional classname
+     */
+    className?: string;
+    /**
+     * Value for input field
+     */
+    value?: string;
+    /**
+     * Disabled for input field
+     */
+    disabled?: boolean;
+    /**
+     * Makes Time picker readonly when set to "true"
+     */
+    readOnly?: boolean;
+    /**
+     * The switch between mobile and desktop version of Dropdown will be applied automatically, when the prop is not specified.
+     * When the prop is present it must be changed from outside. <br>
+     * Possible values: `'desktop' | 'mobile'`
+     */
+    screenType?: 'desktop' | 'mobile';
+    /**
+     * Preferred positions by priority ['bottom', 'top', 'left', 'right']
+     * default is ['bottom', 'top', 'left', 'right']
+     * if you'd like, you can limit the positions ['top', 'left']
+     */
+    positions: 'bottom' | 'top' | 'left' | 'right' | string[];
```

Comment:
1. The positions prop should not be required.
2. For comment please add the `* Possible values: 'bottom' | 'top' | 'left' | 'right' | string[] `.
3. Please change the type , because the array can contain only  ` 'bottom' | 'top' | 'left' | 'right'`  so not `string[]` it should be `['bottom' | 'top' | 'left' | 'right']`.